### PR TITLE
Bug 1177662 - Move header methods to a separate js file

### PR DESCRIPTION
**Pull request by mancas**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8753c491d8862ba4e89e7f597df4e4cff33b5007

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=651a40e5b1f82240418dcbb00db081f70cd0083e

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=acf26d5c0a0c20f153fb84920035dee189128fdc

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5a774924a3cb4ddefb93ccd051677cf0e03cf96c

**Comment by borjasalguero** [comment]
@mancas Could you rebase this with master? Thanks!

File: apps/communications/contacts/js/header_ui.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+(function(exports) {
+  const SELECT_MODE_CLASS = {
+    'pick' : {
+      'text/vcard' : ['disable-fb-items']
```
Comment (borjasalguero) [inline apps/communications/contacts/js/header_ui.js]: Do we need this? As we have no FB now, probably we can remove this!

File: apps/communications/contacts/js/header_ui.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      this.settings = document.getElementById('view-settings');
+      this.settingsButton = document.getElementById('settings-button');
+      this.header = document.getElementById('contacts-list-header');
+      this.addButton = document.getElementById('add-contact-button');
+      this.editModeTitleElement = document.getElementById('edit-title');
+      this.appTitleElement = document.getElementById('app-title');
+    },
+
+    setSelectModeClass: function(element, activityName, activityType) {
+      var classesByType = SELECT_MODE_CLASS[activityName] || {};
```
Comment (borjasalguero) [inline apps/communications/contacts/js/header_ui.js]: As FB is not available anymore, could we update this method accordingly?

File: apps/communications/contacts/js/header_ui.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                                      {n: count});
+    },
+
+    hideAddButton: function() {
+      this.addButton.classList.add('hide');
+    }
+  };
+
+  exports.HeaderUI.init();
+})(window);
```
Comment (borjasalguero) [inline apps/communications/contacts/js/header_ui.js]: One more line needed! :)

File: apps/communications/contacts/js/activities.js [hunk 4]
```diff
@@ -77,13 +77,29 @@ var ActivityHandler = {
...
+      if (activityName === 'pick' || activityName === 'update') {
+        alternativeTitle = 'selectContact';
+      }
+      var groupsList = document.getElementById('groups-list');
+      HeaderUI.setSelectModeClass(groupsList, activityName,
+                              this.activityDataType);
+      HeaderUI.setupCancelableHeader(alternativeTitle);
+    } else {
+      HeaderUI.setupActionableHeader();
+    }
```
Comment (borjasalguero) [inline apps/communications/contacts/js/activities.js]: Do we need to mix UI and Activities? Why dont we have something as `ActivityHandler.isCancelable().then(function(isCancelable) {..... // UI Code})`?

File: apps/communications/contacts/js/header_ui.js (hunk 2 above)
Comment (mancas) [inline apps/communications/contacts/js/header_ui.js]: I think this method is not needed anymore.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=08bd23476d92e79e8adf3877622fc66f5a1fa49d

File: apps/communications/contacts/js/activities.js [hunk 5]
```diff
@@ -77,13 +77,25 @@ var ActivityHandler = {
     return this.currentlyHandling && list.indexOf(this.activityName) === -1;
   },
 
+  isCancelable: function() {
+    return new Promise(resolve => {
+      if (this.currentlyHandling) {
+        resolve(true);
+      } else {
+        resolve(false);
```
Comment (borjasalguero) [inline apps/communications/contacts/js/activities.js]: Could we call `resolve(!!this.currentlyHandling)`?

File: apps/communications/contacts/js/header_ui.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+          alternativeTitle = 'selectContact';
+        }
+        this.setupCancelableHeader(alternativeTitle);
+      } else {
+        this.setupActionableHeader();
+      }
+    }
+  };
+
+  exports.HeaderUI.init();
```
Comment (borjasalguero) [inline apps/communications/contacts/js/header_ui.js]: Do we need to export `init`?

File: apps/communications/contacts/js/header_ui.js [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    _lastCustomHeaderCallback: null,
+
+    init: function() {
+      this.settings = document.getElementById('view-settings');
+      this.settingsButton = document.getElementById('settings-button');
+      this.header = document.getElementById('contacts-list-header');
+      this.addButton = document.getElementById('add-contact-button');
+      this.editModeTitleElement = document.getElementById('edit-title');
+      this.appTitleElement = document.getElementById('app-title');
+    },
```
Comment (borjasalguero) [inline apps/communications/contacts/js/header_ui.js]: Maybe we can execute this without calling `init`, and we save one step. Wdyt?

Comment (mancas) [inline apps/communications/contacts/js/header_ui.js]: Good idea!

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b4601633a8555320dabfd5bad416197f7674f869

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1cb3b1ec911f4dd2e0521816bdec62ce23f47945

**Pull request merged**